objc/src/main/java/org/robovm/objc/ObjCObject.java

```diff
@@ -318,11 +321,61 @@ public ObjCObjectRef(ObjCObject referent) {
         }
     }
     
+    static class ObjectOwnershipHelper {
+        private static final Map<Long, Object> CUSTOM_OBJECTS = new HashMap<Long, Object>();
+        
+        private static final long retainCount = Selector.register("retainCount").getHandle();
+        private static final long retain = Selector.register("retain").getHandle();
+        private static final long newRetain = Selector.register("original_retain").getHandle();
+        private static final long release = Selector.register("release").getHandle();
+        private static final long newRelease = Selector.register("original_release").getHandle();
+        
+        public static void registerClass(ObjCClass objCClass) {
+            registerCallbackMethod(objCClass.getHandle(), retain, newRetain, "retain", ObjCObject.class, Long.TYPE);
+            registerCallbackMethod(objCClass.getHandle(), release, newRelease, "release", Long.TYPE, Long.TYPE);
+        }
+        
+        private static void registerCallbackMethod (long cls, long selector, long newSelector, String methodName, Class<?>... parameterTypes) {
+            Method method = null;
+            try {
+                method = ObjectOwnershipHelper.class.getDeclaredMethod(methodName, parameterTypes);
```

Comment:
This we should be able to do only once in a `static { ... }` block and pass the `Method` references to `registerCallbackMethod()`instead.